### PR TITLE
make timeout

### PR DESCRIPTION
**Pull request by HoloPollock**
make things timeout after 55 seconds and should return a film

### CODE REVIEW

File: api/index.go [hunk 1]
```diff
@@ -26,8 +26,9 @@ type film struct {
 
 //struct for channel to send film and whether is has finshed a user
 type filmSend struct {
-	film film //film to be sent over channel
-	done bool //if user is done
+	film    film //film to be sent over channel
+	done    bool //if user is done
+	timeout bool //if timeout
```
Comment (qjack001) [inline api/index.go:31]: These comments aren't super insightful

File: api/index.go [hunk 2]
```diff
@@ -149,9 +150,15 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 
 //main scraping function
 func scrapeMain(users []string, intersect bool, ignoreList toIgnore) (film, error) {
+
 	var user int = 0          //conuter for number of users increses by one when a users page starts being scraped decreses when user has finished think kinda like a semaphore
 	var totalFilms []film     //final list to hold all film
 	ch := make(chan filmSend) //channel to send films over
+	go func() {
+		time.Sleep(55 * time.Second)
+		log.Println("timeout")
```
Comment (qjack001) [inline api/index.go:159]: I think you should log something more explicit (like, "WARN: maximum time reached, sending whatever we got").

**Comment by qjack001** [comment]
Also make sure to test it against one of those super long lists

**Comment by HoloPollock** [comment]
Yup that's what I did

**Comment by HoloPollock** [comment]
> Also make sure to test it against one of those super long lists

I'll add the logs in a bit to show

File: api/index.go (hunk 2 above)
Comment (HoloPollock) [reply api/index.go:159]: If you want I can switching logging to https://github.com/uber-go/zap for faster and more structured logging

File: api/index.go (hunk 1 above)
Comment (HoloPollock) [reply api/index.go:31]: Lol fair I'll fix

Comment (HoloPollock) [reply api/index.go]: Let me know what you think now

File: api/index.go (hunk 2 above)
Comment (HoloPollock) [reply api/index.go]: Added better logging let me know if you want other things logged

**Review by qjack001 — DISMISSED** [review]

**Comment by HoloPollock** [comment]
Screen Shot 2021-07-30 at 1 19 30 AM
what new logs look like with time out

**Comment by HoloPollock** [comment]
may undo this last commit I found the issue I run out of memory lol 
Screen Shot 2021-07-30 at 1 30 14 AM

**Comment by HoloPollock** [comment]
so for now the log running thing may work or may not but its better than never work

**Comment by HoloPollock** [comment]
I could add a things that checks memory usage and returns when close but that seems stupid

**Comment by HoloPollock** [comment]
also I could switch logger to not sugared logger to save memory

**Comment by qjack001** [comment]
@HoloPollock does this work now?

**Comment by HoloPollock** [comment]
yes if we don't hit a memory error lol

**Comment by qjack001** [comment]
We can try deploying it and seeing if anything happens at scale.

**Review by qjack001 — APPROVED** [review]

**Pull request closed without merging**